Code/Framework/AzCore/AzCore/Serialization/ObjectStream.cpp

```diff
@@ -1350,6 +1358,10 @@ namespace AZ
                     if (GenericClassInfo* genericClassInfo = sc.FindGenericClassInfo(cd->m_typeId))
                     {
                         element.m_id = genericClassInfo->GetSpecializedTypeId();
+                        if (genericClassInfo->GetGenericTypeId() == GetAssetClassId())
+                        {
+                            element.m_id = GetAssetClassId();
+                        }
```

Comment:
Repeat: The DataElement type id should be set to the specialized type id and the check in LoadClass for the Generic Asset Class Id should be updated to see if the specialized type has generic class info with a generic type id that matches the Generic Asset Class Id